packages/wallet/backend/src/grant/service.ts

```diff
@@ -80,6 +80,32 @@ export class GrantService implements IGrantService {
       args.filter = { identifier: { in: identifiers } }
     }
 
-    return await this.rafikiAuthService.listGrantsWithPagination(args)
+    const grants = await this.rafikiAuthService.listGrantsWithPagination(args)
+    this.parseIntervals(grants.grants.edges.map((it) => it.node))
+    return grants
```

Comment:
Usage of the function if it is implemented without mutating grants directly:

```suggestion
   const grants = await this.rafikiAuthService.listGrantsWithPagination(args)
  const parsedGrants = this.parseIntervals(grants.grants.edges.map((it) => it.node))
  return { ...grants, grants: { ...grants.grants, edges: parsedGrants.map(grant => ({ node: grant })) } }
```